script/README.md

```diff
@@ -0,0 +1,73 @@
+# OpenSpace JavaScript API
+
+## TypeScript Declaration Generator
+
+This folder contains a helper Python script that automatically generates a TypeScript declaration file (`d.ts`) for the OpenSpace Javascript API. This allows developers using TypeScript to have syntax highlighting, IntelliSense, and type definitions for the entire API, including OpenSpace Lua functions.
+
+### Why this exists
+
+Instead of publishing the `openspace-api-js.d.ts` file to the @types namespace on npm or the DefinitelyTyped repository, we keep it in this repository. Since the OpenSpace Lua API updates frequently, maintaining control over the TypeScript declaration file ensures that changes can be reflected quickly without going through a lengthy approval process.
+
+## Usage
+
+1.  Copy the `openspace-api-js.d.ts` file from the `declaration` folder to your project's `src` folder.
+2.  Ensure that your TypeScript configuration (`tsconfig.json`) includes the copied `.d.ts`file in `compilerOptions` if necessary.
+
+## Generating an Updated TypeScript Declaration File
+
+To generate an updated `openspace-api-js.d.ts` file:
```

Comment:
```suggestion
The TypeScript declaration file needs to match the Lua API of the OpenSpace version you plan on using for your project. Note that the file in this repository might not be up to date with the latest OpenSpace master, or even released version. 

To generate an updated `openspace-api-js.d.ts` file based on an OpenSpace installation, follow the following steps:
```
? Seemed like some more details could be useful here. But maybe the caveat of "the file in this repo may not be updated in the same rate as openspace" should be mentioned among the caveats as well?